pages/snake/index.js

```diff
@@ -0,0 +1,361 @@
+import Child from "@/components/SnakeGame/Child/Child";
+import Food from "@/components/SnakeGame/Food/Food";
+import Player from "@/components/SnakeGame/Player/Player";
+import StyledLink from "@/components/StyledLink/StyledLink";
+import { useEffect, useState } from "react";
+import styled from "styled-components";
+
+const StyledSnakePage = styled.main`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: center;
+  gap: 30px;
+  width: 100vw;
+  height: 100vh;
+  background: #f1f3c2;
+  @media screen and (min-width: 600px) {
+    gap: 5vh;
+  }
+  @media screen and (min-width: 900px) {
+    gap: 10vh;
+  }
+`;
+
+const StyledGameField = styled.div`
+  display: grid;
+  place-content: center;
+  width: 300px;
+  min-height: 300px;
+  background: lightgreen;
+  border: 2px solid #000000;
+  border-radius: 20px;
+  position: relative;
+  border: 5px solid #000000;
+
+  @media screen and (min-width: 600px) {
+    transform: scale(1.2);
+  }
+  @media screen and (min-width: 900px) {
+    transform: scale(1.5);
+  }
+`;
+
+const StyledScoreContainer = styled.div`
+  display: flex;
+  justify-content: space-between;
+  width: 300px;
+  height: 100px;
+  font-size: 1.2rem;
+  font-weight: 800;
+`;
+
+const StyledButtonContainer = styled.div`
+  display: flex;
+  justify-content: space-between;
+  margin-top: 40px;
+`;
+
+const StyledIndicatorContainer = styled.div`
+  position: absolute;
+  left: -175px;
+  top: 100px;
+
+  @media screen and (min-width: 600px) {
+    left: -200px;
+  }
+
+  h3 {
+    rotate: calc(90deg);
+  }
+
+  article {
+    rotate: calc(270deg);
+    width: 300px;
+    justify-content: center;
+  }
+`;
+
+const StyledGameOver = styled.h1`
+  position: absolute;
+  bottom: 25vh;
+`;
+
+const StyledHowToPlay = styled.div`
+  display: grid;
+  place-content: center;
+  gap: 20px;
+  background: var(--secondary-color);
+  font-size: 0.8rem;
+  line-height: 2;
+  width: 100%;
+  padding: 20px;
+  border-radius: 20px;
+  position: absolute;
+  top: -20px;
+  z-index: 2;
+
+  @media screen and (min-width: 900px) {
+    display: block;
+    width: 120px;
+    right: -140px;
+    top: 20px;
+    font-size: 0.6rem;
+    background: none;
+    padding: 0;
+    line-height: 1.5;
+  }
+
+  @media screen and (min-width: 1200px) {
+    width: 200px;
+    font-size: 0.8rem;
+    right: -230px;
+  }
+`;
+
+export default function SnakeGame({ currentPet, onUpdatePetIndicator }) {
+  const [gameOn, setGameOn] = useState(true);
+  const [playerPosition, setPlayerPosition] = useState({ x: 140, y: 140 });
+  const [children, setChildren] = useState([]);
+  const [foodPosition, setFoodPosition] = useState({});
+  const [direction, setDirection] = useState("");
+  const [prevDirection, setPrevDirection] = useState("");
+  const [score, setScore] = useState(0);
+  const [highscore, setHighscore] = useState(0);
+  const [instructions, setInstructions] = useState(false);
+
```

Comment:
The state management could be simplified to make the code more readable, reduce repetitive logic, and improve maintainability. 

For example you have several pieces of state  that can be grouped together into a single object: `playerPosition`, `foodPosition`, `score`, and `direction` could all be part of a `gameState` `object`.

```js
const [gameState, setGameState] = useState({
  gameOn: true,
  playerPosition: { x: 140, y: 140 },
  children: [],
  foodPosition: generateNewFoodPosition(),
  direction: "",
  prevDirection: "",
  score: 0,
  highscore: 0,
  instructions: false,
});
``` 
Instead of calling multiple state setters, use a single function to update the whole state 😲. For example, you can create a helper function to update the `gameState` like this:


```js
function updateGameState (newState) {
  setGameState((prevState) => ({
    ...prevState,
    ...newState,
  }));
};
```

> 💡 This is an improvement suggestion that I highly recommend, as it offers a great learning opportunity when it comes to state management. However, you don't have to implement it right away. You can add it to your to-do list and tackle it when you're ready. The choice is yours!

Reply:
I tried restructuring my code and failed miserably. 🥲 I'll definitely come back to this though! For now, I only combined the score + highscore and direction + prevDirection into one state each.